packages/grid/_modules_/grid/hooks/features/virtualization/useVirtualColumns.ts

```diff
@@ -115,17 +117,22 @@ export const useVirtualColumns = (
         lastDisplayedIdx > 0 && (diffLast > tolerance || diffFirst > tolerance);
 
       let newRenderedColState: RenderColumnsProps | null =
-        renderedColRef.current != null ? { ...renderedColRef.current } : null;
-
-      const lastVisibleIndex = visibleColumns.length > 0 ? visibleColumns.length - 1 : 0;
-      if (renderNewColState || newRenderedColState == null) {
+        renderedColRef.current !== null ? { ...renderedColRef.current } : null;
+
+      const newLastVisibleIndex = getLastVisibleColIndex(visibleColumns);
```

Comment:
I'm not sure I understand this logic. 
It seems that all we need to do is to generate a new state when we update the visible columns.
Can't we have the same issue on the first col index?

Reply:
No there aren't any problems with the first col (or at least I didn't manage to find any). Also, that helper that I created is no longer needed.